core/src/main/java/org/elasticsearch/index/query/MoreLikeThisQueryBuilder.java

```diff
@@ -944,23 +938,12 @@ private static void handleExclude(BooleanQuery boolQuery, List<Item> likeItems)
     }
 
     @Override
-    public QueryValidationException validate() {
-        QueryValidationException validationException = null;
-        if (likeTexts.isEmpty() && likeItems.isEmpty()) {
-            validationException = addValidationError("requires 'like' to be specified.", validationException);
-        }
-        if (fields != null && fields.isEmpty()) {
-            validationException = addValidationError("requires 'fields' to be specified", validationException);
-        }
-        return validationException;
-    }
-
-    @Override
     protected MoreLikeThisQueryBuilder doReadFrom(StreamInput in) throws IOException {
-        MoreLikeThisQueryBuilder moreLikeThisQueryBuilder = new MoreLikeThisQueryBuilder((List<String>) in.readGenericValue());
-        moreLikeThisQueryBuilder.likeTexts = (List<String>) in.readGenericValue();
+        List<String> fields = (List<String>) in.readGenericValue();
+        List<String> likeTexts = (List<String>) in.readGenericValue();
```

Comment:
I wonder why we use readGenericValue to read a list of strings.... maybe we can move to array here now and use readStringArray?